src/test/unit/services/collections.spec.ts

```diff
@@ -0,0 +1,393 @@
+import { CacheService } from "@multiversx/sdk-nestjs-cache";
+import { TokenUtils } from "@multiversx/sdk-nestjs-common";
+import { Test } from "@nestjs/testing";
+import { ApiConfigService } from "src/common/api-config/api.config.service";
+import { AssetsService } from "src/common/assets/assets.service";
+import { IndexerService } from "src/common/indexer/indexer.service";
+import { PersistenceService } from "src/common/persistence/persistence.service";
+import { PluginService } from "src/common/plugins/plugin.service";
+import { CollectionService } from "src/endpoints/collections/collection.service";
+import { CollectionFilter } from "src/endpoints/collections/entities/collection.filter";
+import { NftCollection } from "src/endpoints/collections/entities/nft.collection";
+import { NftCollectionDetailed } from "src/endpoints/collections/entities/nft.collection.detailed";
+import { EsdtAddressService } from "src/endpoints/esdt/esdt.address.service";
+import { EsdtService } from "src/endpoints/esdt/esdt.service";
+import { NftType } from "src/endpoints/nfts/entities/nft.type";
+import { CollectionRoles } from "src/endpoints/tokens/entities/collection.roles";
+import { TokenAssetStatus } from "src/endpoints/tokens/entities/token.asset.status";
+import { VmQueryService } from "src/endpoints/vm.query/vm.query.service";
+
+describe('CollectionService', () => {
+  let service: CollectionService;
+  let indexerService: IndexerService;
+  let esdtAddressService: EsdtAddressService;
+  let assetsService: AssetsService;
+
+  const indexerCollectionMock = {
```

Comment:
based on last discussion on mocks, this should be a stub :)

Reply:
will be change in a stub